index.bs

```diff
@@ -2234,26 +2235,34 @@ The [=remote end steps=] with |command parameters| are:
   1. Let |type| be the value of the <code>type</code> field of
      |command parameters|.
 
+  1. Let |reference context id| be the value of the <code>context</code> field
+     of |command parameters|, if present, or null otherwise.
+
+  1. If |reference context id| is not null, let |reference context| be the
+     result of [=trying=] to [=get a browsing context=] with
+     |reference context id|. Otherwise let |reference context| be null.
+
   <!-- This is based on step 5 of https://w3c.github.io/webdriver/#new-window,
        but without using the "current browsing context" concept. -->
   1. Create a new [=top-level browsing context=] by running the [=window open
      steps=] with <var ignore>url</var> set to "<code>about:blank</code>",
      <var ignore>target</var> set to the empty string, and
      <var ignore>features</var> set to "<code>noopener</code>". This must be
      done without invoking the [=/focusing steps=] for the created browsing
-     context. If |type| is "<code>tab</code>", and the implementation supports
-     multiple browsing contexts in the same OS window, the new browsing context
-     should share an OS window with any other [=top-level browsing context=]. If
-     |type| is "<code>window</code>", and the implementation supports multiple
-     browsing contexts in separate OS windows, the created browsing context
-     should be in a new OS window. In all other cases the details of how the
-     browsing context is presented to the user are implementation defined.
-
-     Issue: HTML's [=window open steps=] depend on the "entry global object"
-     and WebDriver's <a href="https://w3c.github.io/webdriver/#new-window">New
-     Window</a> command depends on the "current browsing context". Given that we
-     don't have any such implicit context here, what OS window should new tabs
-     be opened in?
+     context. Which OS window the new [=/browsing context=] is created in
+     depends on |type| and |reference context|:
+
+     * If |type| is "<code>tab</code>" and the implementation supports
+       multiple browsing contexts in the same OS window, the new browsing context
+       should share an OS window with |reference context| if not null, and any
```

Comment:
Can we require anything beyond sharing a window? For example, can we require that the new browsing context is directly after _reference context_ in a list of tabs? If so, should we call the parameter something less generic than `context`, like `afterContext`?

Reply:
Well we're not currently exposing the tab ordering in any way, but maybe we should say something like, 
```
* If |type| is "<code>tab</code> and the implementation supports multiple browsing contexts in the same OS window:
   * The new browsing context should re-use an existing OS window, if any.
   * If |reference context| is not null, the should be the window containing |reference context|, if any. If the top level browsing contexts inside an OS window have a definite ordering, the new created top level browsing context should be immediately after |reference context|'s [=top level browsing context=] in that ordering.
 ```
 
 I don't know if the ordering thing also makes sense for OS windows? I don't really know if the concept of OS windows having an order exists anywhere, or if you can choose where new windows appear in that ordering. In any case it seems pretty hard to observe.

Reply:
In Marionette we have the feature to open a new context (tab and window) in the background. The ordering is then checked, but I doubt that this is something we actually would have to offer via WebDriver BiDi. See also the recently closed issue https://github.com/whatwg/html/issues/5872.

Reply:
As a start, I think it makes sense to offer the same guarantees that the classic New Window command has, which is just that the new context shares the same window if requested. There's some other relevant comments in #151.

Reply:
I've pushed a change using James's wording.